pkg/factory/risingwave_object_factory.go

```diff
@@ -1248,6 +1393,40 @@ func buildUpgradeStrategyForStatefulSet(strategy risingwavev1alpha1.RisingWaveUp
 	}
 }
 
+func buildUpgradeStrategyForAdvancedStatefulSet(strategy risingwavev1alpha1.RisingWaveUpgradeStrategy) kruiseappsv1beta1.StatefulSetUpdateStrategy {
+	advancedStatefulSetUpgradeStrategy := kruiseappsv1beta1.StatefulSetUpdateStrategy{}
+	advancedStatefulSetUpgradeStrategy.Type = appsv1.RollingUpdateStatefulSetStrategyType
+	advancedStatefulSetUpgradeStrategy.RollingUpdate = &kruiseappsv1beta1.RollingUpdateStatefulSetStrategy{
+		MaxUnavailable: rollingUpdateOrDefault(strategy.RollingUpdate).MaxUnavailable,
+	}
+	if strategy.InPlaceUpdateStrategy != nil {
+		advancedStatefulSetUpgradeStrategy.RollingUpdate.InPlaceUpdateStrategy = strategy.InPlaceUpdateStrategy.DeepCopy()
+	}
+	if rollingUpdateOrDefault(strategy.RollingUpdate).Partition != nil {
+		// Change a percentage to an integer, partition only accepts int pointers for advanced stateful sets
```

Comment:
Be reminded to add a check in the validating webhook.

Reply:
roger